scipy/optimize/_lsq/least_squares.py

```diff
@@ -439,6 +442,27 @@ def least_squares(
         * 1 : display a termination report.
         * 2 : display progress during iterations (not supported by 'lm'
           method).
+          
+    callback : None or callable, optional
+        Callback function that is called by the algorithm on each iteration. 
+        This can be used to print or plot the optimization results at each
+        step, and to stop the optimization algorithm based on some user-defined
+        condition.  Only implemented for the `trf` and `dogbox` methods.
+        
+        The signature is ``callback(intermediate_result: OptimizeResult)``
+        
+        `intermediate_result is a `scipy.optimize.OptimizeResult` 
+        which contains the intermediate results of the optimization at the 
+        current iteration.
+        
+        The callback also supports a signature like: ``callback(x)``
+
+        Introspection is used to determine which of the signatures is invoked.
+       
+        If the `callback` function raises `StopIteration` or returns `True`,
+        the optimization algorithm will stop and return with status code -2.
+        
+        .. versionadded:: 1.14.2
```

Comment:
minor point -- we don't normally add new features in point releases, so if this gets appropriately reviewed/merged on time, this would be version `1.15.0` for the next feature release